WordPress/Classes/Services/BlogService.m

```diff
@@ -638,22 +638,33 @@ - (OptionsHandler)optionsHandlerWithBlogObjectID:(NSManagedObjectID *)blogObject
         [self.managedObjectContext performBlock:^{
             Blog *blog = (Blog *)[self.managedObjectContext existingObjectWithID:blogObjectID
                                                                            error:nil];
-            if (blog) {
-                blog.options = [NSDictionary dictionaryWithDictionary:options];
-                float version = [[blog version] floatValue];
-                if (version < [MinimumVersion floatValue]) {
-                    if (blog.lastUpdateWarning == nil
-                        || [blog.lastUpdateWarning floatValue] < [MinimumVersion floatValue])
-                    {
-                        // TODO :: Remove UI call from service layer
-                        [WPError showAlertWithTitle:NSLocalizedString(@"WordPress version too old", @"")
-                                            message:[NSString stringWithFormat:NSLocalizedString(@"The site at %@ uses WordPress %@. We recommend to update to the latest version, or at least %@", @""), [blog hostname], [blog version], MinimumVersion]];
-                        blog.lastUpdateWarning = MinimumVersion;
-                    }
+            if (!blog) {
+                if (completion) {
+                    completion();
                 }
-
-                [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
+                return;
+            }
+            blog.options = [NSDictionary dictionaryWithDictionary:options];
+            //HACK:Sergio Estevao (2015-08-31): Because there is no direct way to
+            // know if a user has permissions to change the options we check if the blog title property is read only or not.
+            if (blog.options[@"blog_title"] &&
+                blog.options[@"blog_title"][@"readonly"]) {
```

Comment:
(Just in case) Maybe switch over to Safe Expectations, for peace of mind?

```
[[blog.options dictionaryForKey:@"blog_title"] numberForKey:@"readonly"]
```